src/main/java/org/opensearch/security/privileges/ActionPrivileges.java

```diff
@@ -476,6 +518,16 @@ PrivilegesEvaluatorResponse providesAnyPrivilege(PrivilegesEvaluationContext con
                 }
             }
 
+            // 4: If plugin is performing the action, check if plugin has permission
+            if (this.usersToActionMatcher.containsKey(context.getUser().getName())) {
+                WildcardMatcher matcher = this.usersToActionMatcher.get(context.getUser().getName());
```

Comment:
I guess this should have unit test coverage in  `ActionPrivilegesTest`

Reply:
Added